src/Actions/helpers/Web3ActionCreator.js

```diff
@@ -14,29 +14,14 @@
 //   return { type: WEB3_REQUEST_SUCCEEDED, name, result }
 // }
 
-export function web3Request(name, args, provider, next) {
-  // This specifically pulls state from the web3 reducer. Smell?
-  let web3 = new Web3(provider)
```

Comment:
If we create a new instance of web3 every time we run a request like we were doing before, we wind up with a memory leak, as each new instance of web3 will set up subscription event handlers.

It's possible that there's some way to tell web3 to clean up after itself and unregister its event handlers, etc, but from memory I don't think that there is. Further, if such a mechanism existed and if we were using it here, we'd lose the new block header subscription, because it'd be cleaned up after the `eth_subscribe` request finished.

For better or worse, we should move toward one long-lived instance of Web3 per `provider`.